django_stomp/services/producer.py

```diff
@@ -56,35 +73,46 @@ def send(self, body: dict, queue: str, headers=None, persistent=True, attempt=10
 
         self._send_to_broker(send_data, how_many_attempts=attempt)
 
-    def _build_final_headers(self, queue: str, headers: Dict, persistent: bool):
+    def _build_final_headers(self, queue: str, headers: Dict, persistent: bool) -> Dict:
         """
         Builds the message final headers. Removes unsafe or broker-reserved headers.
+        Standard headers values override headers values to reduce possible errors.
         """
-        correlation_id = current_request_id() if current_request_id() != NO_REQUEST_ID else uuid.uuid4()
+        if headers is None:
+            headers = {}
```

Comment:
**question:** The method signature says that `headers` is a dictionary. Should we expect that a `None` will be given?

Reply:
Yes, it can be `None` as the `send()` method can be invoked without supplying the `headers` parameter which in turn will lead to the `None` possibility! Hehe, gonna update the typehint!